database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Models/PostViewModel.swift

```diff
@@ -25,16 +25,19 @@ class PostViewModel: ObservableObject, Identifiable {
   @Published var body: String
   @Published var starCount: Int
   @Published var userIDsStarredBy: [String: Bool]
```

Comment:
I think I mentioned this before, but would like to remind. `userIDsStarredBy` is an implementation detail and should not be exposed to View. We should make `isStarred` property stored and published instead and update it appropriately.

Not a blocker for this PR, but should be addressed eventually.

Reply:
Currently the `isStarred` property is determined based on the `userIDsStarredBy` and it changes based on who is the current logged in user. I imagine if we store the `isStarred` property for each post instead this value for each post would change based on the current logged in user.

Reply:
Just to make sure we are on the same page. I don't think we need to change database schema or remove `userIDsStarredBy` property entirely. But I think we should make `userIDsStarredBy` private. Then for the view to be updated properly we will need to declare `isStarred` as `@Published`. Currently, if I am not mistaken, `@Published` can be applied only to stored properties, so we will have to make a bit of refactoring because currently `isStarred` is a computed property.